lib/samples/control_annotation_sublayer_visibility/control_annotation_sublayer_visibility.dart

```diff
@@ -0,0 +1,231 @@
+                    onPressed: () => setState(() => _settingsVisible = true),
+                    child: const Text('Settings'),
+                  ),
+                ),
+              ],
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            Visibility(
+              visible: !_ready,
+              child: const SizedBox.expand(
+                child: ColoredBox(
+                  color: Colors.white30,
+                  child: Center(child: CircularProgressIndicator()),
+                ),
+              ),
+            ),
+          ],
+        ),
+      ),
+      bottomSheet: _settingsVisible ? buildSettings(context) : null,
+    );
+  }
+
+  // The build method for the Settings bottom sheet.
+  Widget buildSettings(BuildContext context) {
+    return Container(
+      padding: EdgeInsets.fromLTRB(
+        20.0,
+        20.0,
+        20.0,
+        max(
+          20.0,
+          View.of(context).viewPadding.bottom /
+              View.of(context).devicePixelRatio,
+        ),
+      ),
+      child: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          Row(
+            children: [
+              Text(
+                'Settings',
+                style: Theme.of(context).textTheme.titleLarge,
+              ),
+              const Spacer(),
+              IconButton(
+                icon: const Icon(Icons.close),
+                onPressed: () => setState(() => _settingsVisible = false),
+              ),
+            ],
+          ),
+          if (_openLabel != null)
+            Row(
+              children: [
+                Text(
+                  _openLabel!,
+                  style: TextStyle(color: _openLabelColor),
+                ),
+                const Spacer(),
+                Switch(
+                  value: _openSublayer!.isVisible,
+                  onChanged: (value) {
+                    // Set the visibility of the open sub layer.
+                    if (_openSublayer != null) {
+                      setState(() => _openSublayer!.isVisible = value);
+                    }
+                  },
+                ),
+              ],
+            ),
+          if (_closedLabel != null)
+            Row(
+              children: [
+                Text(
+                  _closedLabel!,
+                ),
+                const Spacer(),
+                Switch(
+                  value: _closedSublayer!.isVisible,
+                  onChanged: (value) {
+                    // Set the visibility of the closed sub layer.
+                    if (_closedSublayer != null) {
+                      setState(() => _closedSublayer!.isVisible = value);
+                    }
+                  },
+                ),
+              ],
+            ),
+          Text(
+            _currentScaleLabel ?? '',
+          ),
+        ],
+      ),
+    );
+  }
+
+  void onMapViewReady() async {
```

Comment:
So we're still a little fuzzy on how we want to do error checking for data that has been specially prepared for a sample. If you look at what we did in `manage_features.dart`, that is probably the best approach we have so far. It has a big try/catch in its `onMapViewReady`. In the body of the `try`, it doesn't check for null and it doesn't check for exceptions thrown by `firstWhere`. Any such problems are caught by the two clauses at the bottom -- one for ArcGISException, and one catch-all.

```dart
  void onMapViewReady() async {
    try {
// ... setting up all the data ...
    } on ArcGISException catch (e) {
      showMessageDialog('${e.message}.');
    } on Exception {
      showMessageDialog(
        'There was an error loading the data required for the sample.',
      );
    }
```
I think we should take the same approach here. That allows the code to read a little better, without having the isNotEmpty and null checks slowing things down.